android/src/main/java/com/swmansion/rnscreens/RNScreensPackage.kt

```diff
@@ -51,4 +95,114 @@ class RNScreensPackage : TurboReactPackage() {
+
+        dummyContentView = View(contextWithTheme).apply {
+            layoutParams = CoordinatorLayout.LayoutParams(
+                CoordinatorLayout.LayoutParams.MATCH_PARENT,
+                CoordinatorLayout.LayoutParams.MATCH_PARENT
+            )
+        }
+
+        coordinatorLayout.apply {
+            addView(appBarLayout)
+            addView(dummyContentView)
+        }
+    }
+
+    /**
+     * Triggers layout pass on dummy view hierarchy, taking into consideration selected
+     * ScreenStackHeaderConfig props that might have impact on final header height.
+     *
+     * @param fontSize font size value as passed from JS
+     * @return header height in dp as consumed by Yoga
+     */
+    private fun computeDummyLayout(fontSize: Int): Float {
+        if (!::coordinatorLayout.isInitialized) {
+            Log.e(TAG, "[RNScreens] Attempt to access dummy view hierarchy before it is initialized")
+            return 0.0f
+        }
+
+        val topLevelDecorView = requireActivity().window.decorView
+
+        // These dimensions are not accurate, as they do include status bar & navigation bar, however
+        // it is ok for our purposes.
+        val decorViewWidth = topLevelDecorView.width
+        val decorViewHeight = topLevelDecorView.height
+
+        val widthMeasureSpec = MeasureSpec.makeMeasureSpec(decorViewWidth, MeasureSpec.EXACTLY)
+        val heightMeasureSpec = MeasureSpec.makeMeasureSpec(decorViewHeight, MeasureSpec.EXACTLY)
+
+        val textView = ScreenStackHeaderConfig.findTitleTextViewInToolbar(toolbar)
+        textView?.textSize = if (fontSize != FONT_SIZE_UNSET) fontSize.toFloat() else defaultFontSize
+
+        coordinatorLayout.measure(widthMeasureSpec, heightMeasureSpec)
+
+        // It seems that measure pass would be enough, however I'm not certain whether there are no
+        // scenarios when layout violates measured dimensions.
+        coordinatorLayout.layout(0, 0, decorViewWidth, decorViewHeight)
+
+        return PixelUtil.toDIPFromPixel(appBarLayout.height.toFloat())
+    }
+
+    private fun requireReactContext(): ReactApplicationContext {
+        return requireNotNull(reactContextRef.get()) { "[RNScreens] Attempt to require missing react context" }
+    }
+
+    private fun requireActivity(): Activity {
+        return requireNotNull(requireReactContext().currentActivity) { "[RNScreens] Attempt to use context detached from activity" }
+    }
+
+    companion object {
+        const val TAG = "RNScreensPackage"
+        const val LIBRARY_NAME = "react_codegen_rnscreens"
+
+        const val FONT_SIZE_UNSET = -1
+
+        // We access this field from C++ layer, through `getInstance` method.
+        // We don't care what instance we get access to as long as it has initialized
+        // dummy view hierarchy.
+        private var WEAK_THIS = WeakReference<RNScreensPackage>(null)
+
+        @JvmStatic
+        fun getInstance(): RNScreensPackage? {
+            return WEAK_THIS.get()
```

Comment:
I'm not sure of how JNI will work here. 

We return the reference here, assuming it is not null, we get it in C++, but who retains it? We hold only a weak ref to self, so if we're dropped by our owner (react package registry) the GC can come in and collect the object while C++ code is running? 

Not sure here on GC thread / the-thread-this-code-is-running-on interaction / execution of C++ code. 

Does the JNI guarantee validity of the returned reference we get in C++ layer for the scope of whole C++ function? My worries are that we do not use "JNI function" there, we just call into JVM from regular C++ code.

Anyway, *I do not think we should care here*, as **WEAK_THIS** lifecycle is tied directly to the application (react context) lifecycle, thus in any not *abnormal* scenario we should be completely fine.